caddyhttp/basicauth/setup_test.go

```diff
@@ -70,6 +70,30 @@ md5:$apr1$l42y8rex$pOA2VJ0x/0TwaFeAF9nX61`
 		}`, false, "pwd", []Rule{
 			{Username: "user", Resources: []string{"/resource1", "/resource2"}},
 		}},
+		{`basicauth user pwd {
+			path /resource1
+			path /resource2
+		}`, false, "pwd", []Rule{
+			{Username: "user", Resources: []string{"/resource1", "/resource2"}},
+		}},
+		{`basicauth user pwd {
+			/resource1
+			realm "Secure resources"
+			path /resource2
```

Comment:
That was my first approach. The problem (not insurmountable) is that when there are three arguments, the intent could be 
```
basicauth path user pwd
```
or
```
basicauth user pwd realm {
  resource 1
  ...
  resource N
}
```
That is, it depends on whether the directive line precedes a nested block. After I had coded it that way, the whole thing seemed fragile and not very future-proof. That's when I decided the use of subdirectives seemed cleaner. I even thought of supporting subdirectives for username and password so the basicauth directive could be free of arguments. I'm fine with reverting to the optional third/fourth argument, but I think that if this directive is ever extended again, it could get messy.